### PR TITLE
Pass ethereum account

### LINKED ISSUE #148: Pass the Ethereum worker address to be used in Ethereum transactions

### PR DESCRIPTION
**Pull request by lenak25**
Fixes #148

### CODE REVIEW

File: src/ethereum/EnigmaContractWriterAPI.js [hunk 1]
```diff
@@ -8,8 +8,12 @@ const config = require('./config.json');
 const EMPTY_HEX_STRING = '0x'; // This is the right value to pass an empty value to the contract, otherwise we get an error
 
 class EnigmaContractWriterAPI extends EnigmaContractReaderAPI {
-  constructor(enigmaContractAddress, enigmaContractABI, web3, logger) {
+  constructor(enigmaContractAddress, enigmaContractABI, web3, logger, walletAddress) {
     super(enigmaContractAddress, enigmaContractABI, web3, logger);
+    this._defaultTrxOptions = config.default;
+    if (walletAddress !== undefined && walletAddress !== null) {
```
Comment (elichai) [inline src/ethereum/EnigmaContractWriterAPI.js]: shouldn't you just do `if (walletAddress)`?

Comment (lenak25) [reply src/ethereum/EnigmaContractWriterAPI.js]: Fixed

**Comment by lenak25** [comment]
@elichai, please review :)

File: src/ethereum/EnigmaContractAPIBuilder.js [hunk 2]
```diff
@@ -83,7 +92,8 @@ class EnigmaContractAPIBuilder {
       this.config = defaultsDeep(config, this.config);
     }
     if (this.config.enigmaContractABI === undefined) {
-      const EnigmaContractJson = require(path.join(this.config.truffleDirectory, 'build/contracts/Enigma.json'));
+      const rawdata = fs.readFileSync((path.join(this.config.truffleDirectory, 'build/contracts/Enigma.json')));//require(path.join(truffleDirectory, 'build/contracts/Enigma.json'));
```
Comment (elichai) [inline src/ethereum/EnigmaContractAPIBuilder.js]: Why the change from `require` to `fs.readFileSync`?

Comment (lenak25) [reply src/ethereum/EnigmaContractAPIBuilder.js]: Because it is only loaded once and if the file changes (which happens between tests) it doesn't updated..
